### ISSUE TITLE
question about proxy

### ISSUE THREAD

**Issue by lzl124631x** [vuejs-templates/webpack#1173]
Hi,

I'm trying to proxy all requests `/api/*` to `localhost:3000/api/*`. No luck for the moment.

According to [webpack-dev-server](https://webpack.github.io/docs/webpack-dev-server.html), I tried to add the following snippet in `dev` object of file `config/index.js`. But it didn't take effect.
```
proxyTable: {
      '/api': {
        target: 'http://localhost:3000',
        secure: false,
        changeOrigin: true // No matter this is true or false
      }
    }
```
The browser kept failing with 404 for API requests e.g. "http://localhost:8080/api/login".

Questions:
1. How to correctly proxy all requests from `localhost/api/*`(front-end) to `localhost:3000/api/*`(back-end)?
2. I found that I will always land on `index.html` with arbitrary link e.g. `http://localhost:8080/123/abc/!@$#/`. Why?

**Comment by LinusBorg** [comment]
1. `target: 'http://localhost:3000/api',`
2. That's because we have set the historyFallback option of the webpack)dev-server, which is required to make vue-router work in history mode. You can read more about that in the router docs. 

For further such general questions, I would recommend to visit forum.vuejs.org or chat.vuejs.org

**Comment by lzl124631x** [comment]
Hi @LinusBorg  Thanks for your reply!

The first one actually should be `target: 'http://localhost:3000` according to [dev-server](https://webpack.js.org/configuration/dev-server/).

My problem was that, after changing the `webpack.dev.config.js`, I have to `ctrl+c` to shut down the dev-server and restart `npm run dev`. **The old node application is NOT shut down along with it** and occupies port `8080`; and the new node application starts at port `8081` (I didn't notice that!). So if I visit `localhost:8080` I got 404; if `localhost:8081` I got 200.
